bridge-deploy/bridge_deploy/cli.py

```diff
@@ -0,0 +1,276 @@
+    help="The initial gas price on the home network in WEI",
+    type=int,
+    default=1_000_000_000,
+)
+
+required_block_confirmations_option = click.option(
+    "--required-block-confirmations",
+    help="The number of blocks to wait for finality",
+    type=int,
+    default=4,
+)
+
+foreign_daily_limit_option = click.option(
+    "--foreign-daily-limit",
+    help="The daily transfer limit for the foreign bridge in WEI",
+    type=int,
+    default=15_000_000_000_000_000_000_000_000,
+)
+
+foreign_max_per_tx_option = click.option(
+    "--foreign-max-per-tx",
+    help="The maximum transfer limit for one transaction in WEI",
+    type=int,
+    default=750_000_000_000_000_000_000_000,
+)
+
+
+@click.group()
+def main():
+    pass
+
+
+@main.command(
+    short_help="Deploys the token bridge on the home network and initializes all contracts."
+)
+@keystore_option
+@validator_set_address_option
+@home_daily_limit_option
+@home_max_per_tx_option
+@home_min_per_tx_option
+@home_gas_price_option
+@required_block_confirmations_option
+@block_reward_address_option
+@foreign_daily_limit_option
+@foreign_max_per_tx_option
+@owner_address_option
+@gas_option
+@gas_price_option
+@nonce_option
+@auto_nonce_option
+@jsonrpc_option
+def deploy_home(
+    keystore: str,
+    jsonrpc: str,
+    gas: int,
+    gas_price: int,
+    nonce: int,
+    auto_nonce: bool,
+    validator_set_address,
+    home_daily_limit,
+    home_max_per_tx,
+    home_min_per_tx,
+    home_gas_price,
+    required_block_confirmations,
+    block_reward_address,
+    foreign_daily_limit,
+    foreign_max_per_tx,
+    owner_address,
+) -> None:
+
+    web3 = connect_to_json_rpc(jsonrpc)
+    private_key = retrieve_private_key(keystore)
+
+    nonce = get_nonce(
+        web3=web3, nonce=nonce, auto_nonce=auto_nonce, private_key=private_key
+    )
+    transaction_options = build_transaction_options(
+        gas=gas, gas_price=gas_price, nonce=nonce
+    )
+
+    dir(transaction_options)
+
+    deployment_result = deploy_home_bridge_contract(
+        web3=web3, transaction_options=transaction_options, private_key=private_key
+    )
+
+    click.echo(f"HomeBridge address: {deployment_result.home_bridge.address}")
+    click.echo(f"  deployed at block #{deployment_result.home_bridge_block_number}")
+
+    initialize_home_bridge_contract(
+        web3=web3,
+        transaction_options=transaction_options,
+        home_bridge_contract=deployment_result.home_bridge,
+        private_key=private_key,
+        validator_contract_address=validator_set_address,
+        home_daily_limit=home_daily_limit,
+        home_max_per_tx=home_max_per_tx,
+        home_min_per_tx=home_min_per_tx,
+        home_gas_price=home_gas_price,
+        required_block_confirmations=required_block_confirmations,
+        block_reward_address=block_reward_address,
+        foreign_daily_limit=foreign_daily_limit,
+        foreign_max_per_tx=foreign_max_per_tx,
+        home_bridge_owner=owner_address,
```

Comment:
You can probably pack all the `init` arguments in one variable if that improves readability.

Reply:
How would that look like? Like a Dict? Are the parameters still required then?

Reply:
I would say like a `NamedeTuple` similarly to `transaction_options`
By init arguments I just mean that `validator_contract_address, home_daily_limit, home_max_per_tx, ..., block_reward_address` are essentially all arguements for initialize function of the contract, so they represent one input argument of the `initialize_home_bridge_contract` python function. Not the `web3`, `transactions_options`, etc ...

Reply:
```transaction_options``` is actually a Dict (just checked) and most of the parameters are optional.
Personally I'd leave it as is for now, as all init params are required and serve a specific purpose.
Also, with the removal of the foreign limits & owner, the list isn't that long anymore ;-)
Fine?

Reply:
fine.